File: backup/Twitch/app/src/main/java/com/exact/twitch/ui/fragment/LazyFragment.java
package com.exact.twitch.ui.fragment;

import androidx.fragment.app.Fragment;

public class LazyFragment extends Fragment {

    private boolean isVisible;
    private boolean loaded;

    @Override
    public void setUserVisibleHint(boolean isVisibleToUser) {
        super.setUserVisibleHint(isVisibleToUser);
        if (isVisibleToUser) {
            isVisible = true;
            if (!loaded) {
                getFragmentManager().beginTransaction().detach(this).attach(this).commit();
                loaded = true;
            }
        } else {
            isVisible = false;
        }
    }

    public boolean isFragmentVisible() {
        return isVisible;
    }
}

File: app/src/main/java/com/exact/twitch/ui/fragment/LazyFragment.kt
package com.exact.twitch.ui.fragment

import androidx.fragment.app.Fragment

open class LazyFragment : androidx.fragment.app.Fragment() {

    var isFragmentVisible: Boolean = false
        private set
    private var loaded: Boolean = false

    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
        super.setUserVisibleHint(isVisibleToUser)
        isFragmentVisible = if (isVisibleToUser) {
            if (!loaded) {
                fragmentManager?.beginTransaction()?.detach(this)?.attach(this)?.commit()
                loaded = true
            }
            true
        } else {
            false
        }
    }
}
